src/test/storage/value_segment_test.cpp

```diff
@@ -103,9 +103,11 @@ TEST_F(StorageValueSegmentTest, MemoryUsageEstimation) {
 
   vs_double.append(42.1337);
 
-  EXPECT_EQ(empty_usage_int + sizeof(int) * 2, vs_int.estimate_memory_usage());
-  EXPECT_EQ(empty_usage_double + sizeof(double), vs_double.estimate_memory_usage());
-  EXPECT_GE(vs_str.estimate_memory_usage(), empty_usage_str + 2 * sizeof(pmr_string));
+  EXPECT_EQ(empty_usage_int + sizeof(int) * 2, vs_int.memory_usage(MemoryUsageCalculationMode::Sampled));
+  EXPECT_EQ(empty_usage_double + sizeof(double), vs_double.memory_usage(MemoryUsageCalculationMode::Sampled));
+
+  EXPECT_GE(vs_str.memory_usage(MemoryUsageCalculationMode::Sampled), empty_usage_str + 2 * sizeof(pmr_string));
+  EXPECT_GE(vs_str.memory_usage(MemoryUsageCalculationMode::Full), empty_usage_str + 2 * sizeof(pmr_string));
```

Comment:
I'd still prefer accurate numbers :(

See `TEST_F(BTreeIndexTest, MemoryConsumptionVeryShortStringNoNulls)` for an example

Reply:
I totally agree!

Reply:
Looks like `BTreeIndexImpl<pmr_string>::_add_to_heap_memory_usage` should be replaced by a call to the new implementation, too.

Reply:
I have overseen this test. GE should no longer be required.

Reply:
I don't really like tests such as in the BTree because it's looks like it is accurate by even specializing on the library, but it's actually just testing our own interpretation of the libraries (which is *wrong* here for llvm).

Just as I side note. Will adapt it.

Reply:
Changed the test of the full estimation to EQ.